stdlib/public/core/MemoryLayout.swift

```diff
@@ -0,0 +1,82 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Accesses the memory layout of `T` through its
+/// `size`, `stride`, and `alignment` properties
+public struct MemoryLayout<T> {
+
+  /// Returns the contiguous memory footprint of `T`.
+  ///
+  /// Does not include any dynamically-allocated or "remote" 
+  /// storage. In particular, `MemoryLayout<T>.size`, when 
+  /// `T` is a class type, is the same regardless of how many 
+  /// stored properties `T` has.
+  @_transparent
+  public static var size: Int { return _sizeof(T.self) }
```

Comment:
If you don't mind, could you directly call the builtin functions in this file?  There is no need for an "abstraction" layer in `_sizeof`, `_alignof` etc.